Adding custom primary colors and header logo

ajolipa opened the pull request:
### In this PR
First pass at adding the ability to customize the primary color and add a logo to the header, as well as updating the footer to the new designs and allowing logos to be added to it.

### Notes and questions
For the time being after struggling a bunch to figure out how to set Tailwind colors using asynchronously fetched data, I ended up putting the customization fields in the `settings` collection rather than the `about` collection, so that they end up in the `config.json` file that's already being copied to the `public` folder at build time. This seems to work, but if we'd prefer those fields to be in the `about` collection, then it could probably be done with an additional build script that runs before `astro build` and copies the color info to somewhere Tailwind can find it. I currently don't think there's an easy way around having to rebuild the site if the primary color is changed.

I wasn't sure exactly how much customization we need to build into the footer, so for now I kept it pretty minimal, but of course more toggles can be added for like, should the background be primary or white, or whatever.

### Screenshots

With custom (light) primary color: 
![image](https://github.com/user-attachments/assets/3e9a0ede-f971-44e0-86d6-77f4361f0c3a)

New category in `Settings` collection in Tina: 
![image](https://github.com/user-attachments/assets/6bf75f87-2033-41ae-9f43-4bace891cbd5)

Adding primary color: 
![image](https://github.com/user-attachments/assets/1d371fd7-b64a-44fb-b773-baf91b75121d)

All settings: 
![image](https://github.com/user-attachments/assets/04fc19c6-bacb-4f4a-9c12-022ee63335ab)

dleadbetter commented on src/layouts/Footer.astro:
```diff
@@ -2,90 +2,101 @@
+            href={config.core_data.url}
+            target='_blank'
+          >
+            <Button client:only rounded className='bg-white'>
```
Any reason we're adding the `client:only` directive here?

ajolipa replied:
Yes, the page crashed without it. 😅 Other option would be not to use the `Button` component from `core-data`, but I felt I should use it so that any updates we make to that component apply in the footer as well for consistent aesthetics.

dleadbetter commented on src/layouts/Layout.astro:
```diff
@@ -57,6 +60,7 @@ const tabs = {
   <body
     class={clsx(
       'flex h-full sm:min-h-screen flex-col bg-white text-neutral-dark mx-auto',
+      `primary-text-${primaryText}`,
```
Could we use the conditional class names here rather than constructing the class name?

```
{ "primary-text-white": !useBlackText },
{ "primary-text-black: useBlackText }
```

dleadbetter commented on src/utils/layout.ts:
```diff
@@ -0,0 +1,24 @@
+    const g = g_ratio <= 0.04045 ? g_ratio/12.92 : ((g_ratio + 0.055)/1.055)**2.4;
+    const b = b_ratio <= 0.04045 ? b_ratio/12.92 : ((b_ratio + 0.055)/1.055)**2.4;
+    const l = 0.2126 * r + 0.7152 * g + 0.0722 * b;
+    if (l > 0.179) {
```
Can this be simplified to `return l > 0.179`?

ajolipa replied:
Haha what a great point. 😅

dleadbetter reviewed (commented):
This looks good. Store the branding in `config.json` will make it so that customizations are done by the developer during the deployment, as opposed to the end-user. Any changes to branding will require the site to be re-built. @jamiefolsom - Is this inline with the requirements?

dleadbetter commented on src/layouts/Footer.astro:
```diff
@@ -2,90 +2,101 @@
+            href='/admin/index.html'
+            target='_blank'
+          >
+            <Button client:only rounded className='bg-white'>
```
Will this always be `bg-white`?

ajolipa replied:
Good question. For simplicity for now I just have the `bg-primary` version of the footer, to match the header. There could also be an inverted version with a white background and `primary` buttons, but beyond that, do we want to allow customization of the button color as well as `primary`?

ajolipa commented:
Update: Addressed issues above, plus a few tweaks to the mobile header to bring it in line with the new styles. I also took this opportunity to remove the max width on wide screens, because I had originally put that in by default but in fact the layout of this site is simple enough that it doesn't seem necessary. (The width limit is necessary on some more elaborate sites with background images and such to avoid weird layout spread, but here everything's just in central columns.)
